src/envs/cover.py

```diff
@@ -345,3 +345,372 @@ def __init__(self) -> None:
+        del task  # not used by this render function
+        del action  # not used by this render function
+        fig, ax = plt.subplots(1, 1)
+        # Draw main line
+        plt.plot([-0.2, 1.2], [-0.055, -0.055], color="black")
+        # Draw hand regions
+        hand_regions = self._get_hand_regions(state)
+        for i, (hand_lb, hand_rb) in enumerate(hand_regions):
+            if i == 0:
+                label = "Allowed hand region"
+            else:
+                label = None
+            plt.plot([hand_lb, hand_rb], [-0.08, -0.08], color="red",
+                     alpha=0.5, lw=8., label=label)
+        # Draw hand
+        plt.scatter(state.get(self._robot, "x"),
+                    state.get(self._robot, "y"),
+                    color="r", s=100, alpha=1., zorder=10, label="Hand")
+        lw = 3
+        cs = ["blue", "purple", "green", "yellow"]
+        block_alpha = 0.75
+        targ_alpha = 0.25
+        # Draw blocks
+        for i, block in enumerate(self._blocks):
+            c = cs[i]
+            if state.get(block, "grasp") != -1:
+                lcolor = "red"
+                bx = state.get(self._robot, "x") - state.get(block, "grasp")
+                by = state.get(self._robot, "y")
+                suffix = " (grasped)"
+            else:
+                lcolor = "gray"
+                bx = state.get(block, "x")
+                by = state.get(block, "y")
+                suffix = ""
+            rect = plt.Rectangle(
+                (bx - state.get(block, "width")/2.,
+                 by - state.get(block, "height")),
+                state.get(block, "width"),
+                state.get(block, "height"),
+                linewidth=lw, edgecolor=lcolor, facecolor=c,
+                alpha=block_alpha, label=f"block{i}"+suffix)
+            ax.add_patch(rect)
+        # Draw targets
+        for i, targ in enumerate(self._targets):
+            c = cs[i]
+            rect = plt.Rectangle(
+                (state.get(targ, "x")-state.get(targ, "width")/2.,
+                 -self.target_height/2.),
+                state.get(targ, "width"),
+                self.target_height,
+                linewidth=lw,
+                edgecolor=lcolor,
+                facecolor=c, alpha=targ_alpha, label=f"target{i}")
+            ax.add_patch(rect)
+        grip = state.get(self._robot, "grip")
+        plt.title(f"Grip: {grip:.3f}")
+        plt.xlim(-0.2, 1.2)
+        plt.ylim(-0.25, 0.5)
+        plt.legend()
+        plt.tight_layout()
+        img = utils.fig2data(fig)
+        plt.close()
+        return [img]
+
+    def _create_initial_state(self, rng: np.random.Generator) -> State:
+        # Need to override to account for new low-level state features.
+        # Note: I originally tried to use super(), but ran into issues
+        # because the parent class types and this class types are
+        # actually different (in terms of equality checking).
```

Comment:
nice catch.... do we want to change type equality checking to only be based on the type name, not on the features? fine to leave as is too

Reply:
I think it's good as is. Without this, I may have not realized that the types are different, and that could've caused problems.